lib/VCMIDirs.cpp

```diff
@@ -164,178 +97,111 @@ class VCMIDirsWIN32 final : public IVCMIDirs
 		std::string libraryName(const std::string& basename) const override;
 
 		void init() override;
+
 	protected:
-		bfs::path oldUserDataPath() const;
-		bfs::path oldUserSavePath() const;
+		mutable std::optional<JsonNode> dirsConfig;
+		void loadDirsJsonIfNeeded() const;
+		bfs::path getPathFromConfigOrDefault(const std::string& key, const std::function<bfs::path()>& fallbackFunc) const;
+		std::wstring utf8ToWstring(const std::string& str) const;
 };
 
-void VCMIDirsWIN32::init()
+std::wstring VCMIDirsWIN32::utf8ToWstring(const std::string& str) const
```

Comment:
please check if codebase already has a similar function or a piece of code that does the same thing. IIRC I added something like that for argv handling, but it probably uses CP_ACP.

Reply:
also, this function should be in a respective file, TextOperations if my memory serves me well

Reply:
I have not found another function with same functionality for now. If it's somewhere, will be better of course to reuse it instead adding another one.